src/lib/es5.d.ts

```diff
@@ -1689,12 +1685,12 @@ interface Int8Array {
       * the accumulation. The first call to the callbackfn function provides this value as an argument
       * instead of an array value.
       */
-    reduceRight<U>(callbackfn: (previousValue: U, currentValue: number, currentIndex: number, array: Int8Array) => U, initialValue: U): U;
+    reduceRight<U>(callbackfn: (previousValue: U, currentValue: number, currentIndex: number, array: this) => U, initialValue: U): U;
 
     /**
       * Reverses the elements in an Array.
       */
-    reverse(): Int8Array;
+    reverse(): T;

```

Comment:
Rather than return `T`, either return `this` or return `TypedArray` with a subtyped overload in each specific typed-array constructor (e.g. return `Int32Array` in the `Int32Array` interface).